code/modules/mob/living/simple_animal/bot/mulebot.dm

```diff
@@ -545,11 +541,19 @@
 
 /mob/living/simple_animal/bot/mulebot/Moved() //make sure we always use power after moving.
 	. = ..()
-	if(!cell)
-		return
-	cell.use(cell_move_power_usage)
+
+	cell?.use(cell_move_power_usage)
```

Comment:
i'm not a fan of this since this means if the mulebot is pulled around by a singularity, it'll lose power.

Reply:
Agreed. I shifted this into an overridden `/mob/living/simple_animal/bot/mulebot/bot_step`

The bot should now not-move if it's off and not-move and turn itself off if it doesn't have the cell power to move.

Reply:
Moved to signals and signal handlers.